packages/grpc-native-core/index.d.ts

```diff
@@ -0,0 +1,1176 @@
+     * does not use SSL. This cannot be composed with anything.
+     * @return The insecure credentials object
+     */
+    createInsecure(): ChannelCredentials;
+  };
+
+  /**
+   * Metadata generator function.
+   */
+  export type metadataGenerator = (params: { service_url: string }, callback: (error: Error | null, metadata?: Metadata) => void) => void;
+
+  /**
+   * This cannot be constructed directly. Instead, instances of this class should
+   * be created using the factory functions in `grpc.credentials`
+   */
+  export class ChannelCredentials {}
+
+  /**
+   * This cannot be constructed directly. Instead, instances of this class should
+   * be created using the factory functions in `grpc.credentials`
+   */
+  export class CallCredentials {}
+
+  /**
+   * Google credential from https://github.com/google/google-auth-library-nodejs lib.
+   * The object definition is copied from `ts/lib/auth/credentials.ts`
+   */
+  export interface GoogleCredential {
+      refresh_token?: string;
+      expiry_date?: number;
+      access_token?: string;
+      token_type?: string;
+  }
+
+  /**
+   * Creates a constructor for a client with the given methods, as specified in
+   * the methods argument. The resulting class will have an instance method for
+   * each method in the service, which is a partial application of one of the
+   * `grpc.Client` request methods, depending on `requestSerialize`
+   * and `responseSerialize`, with the `method`, `serialize`, and `deserialize`
+   * arguments predefined.
+   * @param methods An object mapping method names to method attributes
+   * @param serviceName The fully qualified name of the service
+   * @param classOptions An options object.
+   * @return New client constructor, which is a subclass of `grpc.Client`, and
+   * has the same arguments as that constructor.
+   */
+  export function makeGenericClientConstructor(
+    methods: ServiceDefinition,
+    serviceName: string,
+    classOptions: GenericClientOptions,
+  ): typeof Client;
+
+  /**
+   * Options for generic client constructor.
+   */
+  export interface GenericClientOptions {
+    /**
+     * Indicates that the old argument order should be used for methods, with
+     * optional arguments at the end instead of the callback at the end. This
+     * option is only a temporary stopgap measure to smooth an API breakage.
+     * It is deprecated, and new code should not use it.
+     */
+    deprecatedArgumentOrder?: boolean;
+  }
+
+  /**
+   * Create a client with the given methods
+   */
+  export class Client {
+    /**
+     * A generic gRPC client. Primarily useful as a base class for generated clients
+     * @param address Server address to connect to
+     * @param credentials Credentials to use to connect to the server
+     * @param {Object} options Options to apply to channel creation
+     */
+    constructor(address: string, credentials: ChannelCredentials, options?: object)
```

Comment:
By the way, I creaed typings for the options [here](https://github.com/mixer/etcd3/blob/master/src/types/grpc.d.ts#L258)

Reply:
Great, I'll merge them into this file. Thanks!

Reply:
That's good to know; I would prefer not to include those in this PR. There's some complexity with handling the case where new arguments are added, and I think we'll be able to handle that better in the future after some other changes we're planning to make.

Reply:
Roger. We leave them out for now then.